### PR TITLE
Combine frontend and backend into traditional monorepo

### PR DESCRIPTION
**Pull request by ElijahMSmith**
This commit creates one node project that can run either the frontend, backend, or both via one set of node scripts.

**This commit DOES NOT change the contents of any files** other than the `README.md` and `package.json`, which are combined from both folders.

Configuration files are moved to the top-level of the project. In order to migrate your branches, make sure to copy your private file (.env and the firebase SDK configuration) to the top-level of the project, as well as delete the node_modules folder from the frontend and backend folders (and running 'npm i' from the main level after rebasing).

You can run `npm run dev` to start both the frontend and backend together, or just `npm run dev:client` or `npm run dev:server`.

Feel free to comment on the structure here and anything we can improve! I will open subsequent MRs once this is merged to:
1) Add a `.prettierrc` file and reformat everything according to a standard ruleset.
2) Set up Vite config to proxy requests to the backend server. This way we can fetch to `/api/...` instead of needing to explicitly set a URL.

Let me know if there is anything else that would be helpful to include in this setup!

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+		"nodemon": "^2.0.21",
+		"vite": "^5.3.4"
+	},
+	"scripts": {
+		"dev": "concurrently \"npm run dev:server\" \"npm run dev:client\"",
+		"build": "npm run build:client",
+		"dev:server": "nodemon server/server.mjs",
+		"dev:client": "vite",
+		"build:client": "vite build",
+		"preview:client": "vite preview"
```
Comment (mattschmitz) [inline package.json:43]: Thanks for setting these scripts up. I think this will make dev a lot faster

**Review by mattschmitz — APPROVED** [review]
Thanks so much for doing this Eli, you are awesome!

I am fine with this folder structure, I've seen this pattern before. I don't know how you deploy it, but I'm sure there are guides.

Hopefully

**Review by SonaliBedge — APPROVED** [review]
Thank you for setting this up.

**Review by lauradurandv — APPROVED** [review]
Thank you for your directions and explanation. They are very helpful.

**Review by EdithHarrison — APPROVED** [review]

**Pull request merged**